maths/euclidean_distance.py

```diff
@@ -0,0 +1,60 @@
+from typing import Iterable, Union
+
+import numpy as np
+
+Vector = Union[Iterable[float], Iterable[int], np.ndarray]
+VectorOut = Union[np.float64, int, float]
+
+
+def euclidean_distance(vector_1: Vector, vector_2: Vector) -> VectorOut:
+    """
+    Calculate the distance between the two endpoints of two vectors.
+    A vector is defined as a list, tuple, or numpy 1D array.
+    >>> euclidean_distance((0, 0), (2, 2))
+    2.8284271247461903
+    >>> euclidean_distance(np.array([0, 0, 0]), np.array([2, 2, 2]))
+    3.4641016151377544
+    >>> euclidean_distance(np.array([1, 2, 3, 4]), np.array([5, 6, 7, 8]))
+    8.0
+    >>> euclidean_distance([1, 2, 3, 4], [5, 6, 7, 8])
+    8.0
+    """
+    return np.sqrt(np.sum((np.asarray(vector_1) - np.asarray(vector_2)) ** 2))
+
+
+def euclidean_distance_no_np(vector_1: Vector, vector_2: Vector) -> VectorOut:
+    """
+    Calculate the distance between the two endpoints of two vectors without numpy.
+    A vector is defined as a list, tuple, or numpy 1D array.
+    >>> euclidean_distance_no_np((0, 0), (2, 2))
+    2.8284271247461903
+    >>> euclidean_distance_no_np([1, 2, 3, 4], [5, 6, 7, 8])
+    8.0
+    """
+    return sum([(y - x) ** 2 for (y, x) in list(zip(vector_1, vector_2))]) 
+
+
+def benchmark() -> None:
+    """
+    Benchmarks
+    """
+    from timeit import timeit
+
+    print(
+        timeit(
+            "euclidean_distance_no_np([1, 2, 3], [4, 5, 6])",
+            setup="from __main__ import euclidean_distance_no_np",
+            number=10000,
+        )
+    )
+    print(
+        timeit(
+            "euclidean_distance([1, 2, 3], [4, 5, 6])",
+            setup="from __main__ import euclidean_distance",
+            number=10000,
+        )
+    )
+
```

Comment:
Please put this directly in the `if __name__ == "__main__":` block.

Reply:
Also, use the `globals()` dictionary in the `timeit` function like so: `timeit(..., globals=globals())` instead of the `setup`  parameter.

Reply:
One more thing: Maybe print out what those numbers actually mean?
```
$ python euclidean_distance.py
0.019355665999999994
0.12133019300000003
```
This would be pretty confusing when someone runs this script.